compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1983,7 +2007,31 @@ impl<'context> Elaborator<'context> {
             self.file = trait_impl.file_id;
             self.local_module = trait_impl.module_id;
 
-            let trait_id = self.resolve_trait_by_path(trait_impl.trait_path.clone());
+            let trait_type = match &trait_impl.r#trait.typ {
+                UnresolvedTypeData::Named(trait_path, trait_generics, _) => {
+                    TraitType::Named(trait_path.clone(), trait_generics.clone())
+                }
+                UnresolvedTypeData::Resolved(quoted_type_id) => {
+                    let typ = self.interner.get_quoted_type(*quoted_type_id);
+                    if let Type::TraitAsType(trait_id, _, trait_generics) = typ {
+                        TraitType::Trait(*trait_id, trait_generics.clone())
+                    } else {
+                        self.push_err(ResolverError::ExpectedTrait {
+                            span: trait_impl.r#trait.span,
+                        });
```

Comment:
Looks like we could convert `typ` to a string here to keep the "but found" part of the error

Reply:
Done!